libs-scala/resources-akka/src/main/scala/com/digitalasset/resources/akka/AkkaResourceOwnerFactories.scala

```diff
@@ -28,8 +30,23 @@ trait AkkaResourceOwnerFactories[Context] {
   def forCancellable[C <: Cancellable](acquire: () => C): AbstractResourceOwner[Context, C] =
     new CancellableResourceOwner(acquire)
 
-  def forBoundedSourceQueue[T](
-      queueGraph: RunnableGraph[BoundedSourceQueue[T]]
-  )(implicit materializer: Materializer): AbstractResourceOwner[Context, BoundedSourceQueue[T]] =
-    new BoundedSourceQueueResourceOwner[T, Context](queueGraph)
+  def forBoundedSourceQueue[T, U](
```

Comment:
Can we make this one the only supported factory? And change the `BoundedSourceQueueResourceOwner` accordingly to run the queue and return the future?